libindy/src/services/metrics/mod.rs

```diff
@@ -0,0 +1,113 @@
+    }
+    const fn commands_count() -> usize {
+        CommandIndex::VARIANT_COUNT
+    }
+
+    pub fn append_command_metrics(&self, metrics_map: &mut Map<String, Value>) {
+        for index in (0..MetricsService::commands_count()).rev() {
+            let cmd_name = MetricsService::cmd_name(index);
+            metrics_map.insert(format!("{}_{}", cmd_name.as_str(), EXECUTED_COMMANDS_COUNT).as_str().to_string(),
+                               Value::from(self.executed_commands_count.borrow()[index] as usize));
+            metrics_map.insert(format!("{}_{}", cmd_name.as_str(), EXECUTED_COMMANDS_DURATION_MS).as_str().to_string(),
+                               Value::from(self.executed_commands_duration_ms.borrow()[index] as usize));
+            metrics_map.insert(format!("{}_{}", cmd_name.as_str(), QUEUED_COMMANDS_COUNT).as_str().to_string(),
+                               Value::from(self.queued_commands_count.borrow()[index] as usize));
+            metrics_map.insert(format!("{}_{}", cmd_name.as_str(), QUEUED_COMMANDS_DURATION_MS).as_str().to_string(),
+                               Value::from(self.queued_commands_duration_ms.borrow()[index] as usize));
+        }
+    }
+}
+
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_cmd_left_queue() {
+        let metrics_service = MetricsService::new();
+        let index = CommandIndex::IssuerCommandCreateSchema;
+        let durations = [5u128, 2u128, 3u128];
+        let command_count: u128 = durations.len() as u128;
+
+        for duration in durations.iter() {
+            metrics_service.cmd_left_queue(index, *duration);
+        }
+
+        assert_eq!(metrics_service.queued_commands_count.borrow()[index as usize], command_count);
+        assert_eq!(metrics_service.queued_commands_duration_ms.borrow()[index as usize], durations.iter().sum::<u128>());
+        assert_eq!(metrics_service.executed_commands_count.borrow()[index as usize], 0);
+        assert_eq!(metrics_service.executed_commands_duration_ms.borrow()[index as usize], 0);
+    }
+    #[test]
+    fn test_cmd_executed() {
+        let metrics_service = MetricsService::new();
+        let index = CommandIndex::IssuerCommandCreateSchema;
+        let durations = [5u128, 2u128, 3u128];
+        let command_count: u128 = durations.len() as u128;
+
+        for duration in durations.iter() {
+            metrics_service.cmd_executed(index, *duration);
+        }
+
+        assert_eq!(metrics_service.queued_commands_count.borrow()[index as usize], 0);
+        assert_eq!(metrics_service.queued_commands_duration_ms.borrow()[index as usize], 0);
+        assert_eq!(metrics_service.executed_commands_count.borrow()[index as usize], command_count);
+        assert_eq!(metrics_service.executed_commands_duration_ms.borrow()[index as usize], durations.iter().sum::<u128>());
+    }
+    #[test]
+    fn test_append_command_metrics() {
+        let payment_service = MetricsService::new();
```

Comment:
Name `payment_service` looks a bit misleading in my opinion

Reply:
fixed